src/api/v1/users/volunteers/volunteer_logs.ts

```diff
@@ -0,0 +1,201 @@
+        pre: { communityBusiness },
+        query,
+      } = request;
+
+      const since = new Date(query.since);
+      const until = new Date(query.until);
+
+      const logs = await VolunteerLogs.fromUserAtCommunityBusiness(
+        knex,
+        user,
+        communityBusiness,
+        { since, until, limit: query.limit, offset: query.offset }
+      );
+
+      return Promise.all(logs.map(VolunteerLogs.serialise));
+    },
+  },
+
+  {
+    method: 'GET',
+    path: '/users/volunteers/me/volunteer-logs/{logId}',
+    options: {
+      description: 'Read own volunteer logs',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['volunteer_logs-parent:read'],
+        },
+      },
+      validate: {
+        query: { fields: query.fields },
+        params: { logId: id },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request: GetVolunteerLogRequest, h) => {
+      const {
+        server: { app: { knex } },
+        auth: { credentials: { user } },
+        pre: { communityBusiness },
+        params: { logId },
+        query,
+      } = request;
+
+      const log = await VolunteerLogs.getOne(
+        knex,
+        {
+          where: {
+            id: Number(logId),
+            userId: user.id,
+            organisationId: communityBusiness.id,
+            deletedAt: null,
+          },
+          fields: query.fields,
+        }
+      );
+
+      return !log
+        ? Boom.notFound('No log with this id found under this account')
+        : VolunteerLogs.serialise(log);
+    },
+  },
+
+  {
+    method: 'PUT',
+    path: '/users/volunteers/me/volunteer-logs/{logId}',
+    options: {
+      description: 'Delete own volunteer logs',
+      auth: {
+        strategy: 'standard',
+        access: {
+          scope: ['volunteer_logs-parent:write'],
+        },
+      },
+      validate:  {
+        params: { logId: id },
+        payload: {
+          activity: Joi.string(),
+          duration: Joi.object({
+            hours: Joi.number().integer().min(0),
+            minutes: Joi.number().integer().min(0),
+            seconds: Joi.number().integer().min(0),
+          }),
+          startedAt: Joi.date().iso().max('now'),
+        },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request: PutMyVolunteerLogRequest, h) => {
+      const {
+        server: { app: { knex } },
+        auth: { credentials: { user } },
+        payload,
+        pre: { communityBusiness },
+        params: { logId },
+      } = request;
+
+      const log = await VolunteerLogs.getOne(knex, { where: {
+        id: Number(logId),
+        userId: user.id,
+        organisationId: communityBusiness.id,
+        deletedAt: null,
+      }});
```

Comment:
Out of curiosity, have you turned off tslint?

Reply:
No

Reply:
Weird, on my local it adds spaces between closing brackets.

Reply:
Mine too but not in this case, must be an edge case in the parser?